packages/flutter_tools/lib/src/migrations/swift_package_manager_integration_migration.dart

```diff
@@ -557,69 +565,87 @@ $newContent
+    final int nativeTargetStartIndex = lines.indexWhere(
       (String line) => line.trim().startsWith(subsectionLineStart),
       startSectionIndex,
     );
-    if (runnerNativeTargetStartIndex == -1 || runnerNativeTargetStartIndex > endSectionIndex) {
+
+    if (nativeTargetStartIndex == -1 || nativeTargetStartIndex > endSectionIndex) {
       throw Exception(
         'Unable to find PBXNativeTarget for ${_xcodeProject.hostAppProjectName} target.',
       );
     }
 
-    if (runnerNativeTarget.packageProductDependencies == null) {
+    if (nativeTarget.packageProductDependencies == null) {
       // If packageProductDependencies is null, the packageProductDependencies field is missing and must be added.
       const List<String> newContent = <String>[
         '			packageProductDependencies = (',
         '				$_flutterPluginsSwiftPackageProductDependencyIdentifier /* FlutterGeneratedPluginSwiftPackage */,',
         '			);',
       ];
-      lines.insertAll(runnerNativeTargetStartIndex + 1, newContent);
-    } else {
-      // Find the packageProductDependencies field within the Native Target for the Runner target.
-      final int packageProductDependenciesIndex = lines.indexWhere(
-        (String line) => line.trim().contains('packageProductDependencies'),
-        runnerNativeTargetStartIndex,
+      lines.insertAll(nativeTargetStartIndex + 1, newContent);
+      return;
+    }
+
+    // Find the packageProductDependencies field within the Native Target.
+    final int packageProductDependenciesIndex = lines.indexWhere(
+      (String line) => line.trim().contains('packageProductDependencies'),
+      nativeTargetStartIndex,
+    );
+    if (packageProductDependenciesIndex == -1 ||
+        packageProductDependenciesIndex > endSectionIndex) {
+      throw Exception(
+        'Unable to find packageProductDependencies for ${_xcodeProject.hostAppProjectName} PBXNativeTarget.',
+      );
+    }
+    const String newContent =
+        '				$_flutterPluginsSwiftPackageProductDependencyIdentifier /* FlutterGeneratedPluginSwiftPackage */,';
+    lines.insert(packageProductDependenciesIndex + 1, newContent);
+  }
+
+  List<String> _migrateNativeApplicationTargets(List<String> lines, ParsedProjectInfo projectInfo) {
+    if (_areNativeApplicationTargetsMigrated(projectInfo)) {
+      logger.printTrace('PBXNativeTargets already migrated. Skipping...');
+      return lines;
+    }
+
+    final (int startSectionIndex, int endSectionIndex) = _sectionRange('PBXNativeTarget', lines);
+
+    // Apply the migration to each application target within the section.
+    final Iterable<ParsedNativeTarget> applicationNativeTargets = projectInfo.nativeTargets.where(
+      (ParsedNativeTarget target) => target.productType == _kNativeTargetApplicationProductType,
+    );
+
+    if (applicationNativeTargets.isEmpty) {
+      throw Exception(
+        'Unable to find parsed PBXNativeTargets for ${_xcodeProject.hostAppProjectName} project.',
```

Comment:
The old code had `target.` at the end, but that is a bit misleading. It's the `Runner` _project_ that has N Targets.
Because the migrator now handles multiple Targets, it gets more confusing so I updated it here to rectify that.